helpers/removeAllRoomRatesBarCheapest.js

```diff
@@ -0,0 +1,41 @@
+/* jslint node: true */
+'use strict';
+
+var _ = require( 'lodash' );
+
+module.exports = function( dust ) {
+
+	/*
+	* @description Find cheapest hotel and strip rest from api reply.
+	* @param {string} object
+	* @example {@_forEach object=Object /}
+	*/
+
+	// This whole helper is very specific, needs to be rewritten asap.
+	dust.helpers._removeAllRoomRatesBarCheapest = function( chunk, context, bodies, params ) {
+		params = params || {};
+
+		// Use our params
+		var roomRates = params.object.linked.roomRates;
+		var packageRates = params.object.packageRates;
+		// Toss the used ones, so we can return everything else to the template
+		delete params[ 'object' ];
+		
+		var cheapestRoom = { id: 0, price: -1};
+
+		_.forEach( roomRates, function( roomRate ){
```

Comment:
This would probably be better written as a reduce function (something along the lines of)

```
var cheapestRoom = roomRates.reduce( function( currentCheapest, roomRate) {
  return ( roomRate.grossPrice < currentCheapest.price || currentCheapest.price == -1 ) ? roomRate : currentCheapest;
} ,  { id: 0, price: -1 } );

(//i'm not too keen on the  { id: 0, price: -1 } but guess you have a reason for it.

(I've just typed the code here myself so probably made a silly syntax error) 
This is a cool guide to reduce.
https://www.airpair.com/javascript/javascript-array-reduce